node/node.go

```diff
@@ -170,6 +173,14 @@ func (n *Node) serveHttp(ctx context.Context, isSSLDisabled bool, sslEmail strin
 		addPeerHandler(w, r, n)
 	})
 
+	handler.HandleFunc(endpointBlockByNumber, func(w http.ResponseWriter, r *http.Request) {
+		p := strings.Split(r.URL.Path, "/")[1:]
+		if len(p) < 2 {
+			return
+		}
```

Comment:
Thanks for adding the integration test @deferbot !

I see this logic is duplicated twice. Once in the test and once in the node pkg. Can you move this param splitting directly into the http_routes.go -> blockByNumberOrHash()? So we have it in one place.